### PR TITLE
fix(default-config): update default config generation, include insigh…

### PR DESCRIPTION
**Pull request by bitjson**
…t-api and insight-ui

### CODE REVIEW

File: lib/scaffold/default-config.js [hunk 1]
```diff
@@ -24,34 +24,47 @@ function getDefaultConfig(options) {
...
+  var defaultDataDir = path.resolve(defaultPath, './data');
+
+  if (!fs.existsSync(defaultDataDir)) {
+    mkdirp.sync(defaultDataDir);
+  }
+
   if (!fs.existsSync(defaultConfigFile)) {
     var defaultConfig = {
-      network: 'livenet',
+      network: 'testnet',
```
Comment (gabegattis) [inline lib/scaffold/default-config.js]: I think we should default to livenet.

File: lib/scaffold/start.js [hunk 2]
```diff
@@ -86,10 +86,12 @@ function lookInRequirePathConfig(req, service) {
 }
 
 function lookInCwd(req, service) {
+  var location = service.config.cwdRequirePath ? service.config.cwdRequirePath : service.name
```
Comment (gabegattis) [inline lib/scaffold/start.js]: please use jshint (missing semicolon)

File: lib/scaffold/start.js [hunk 3]
```diff
@@ -98,7 +100,7 @@ function lookInBuiltInPath(req, service) {
     var serviceFile = path.resolve(__dirname, '../services/' + service.name);
     return req(serviceFile);
   } catch(e) {
-    console.log(e);
+    if(e.code !== 'MODULE_NOT_FOUND') log.error(e);
```
Comment (gabegattis) [inline lib/scaffold/start.js]: please use jshint (no single-line ifs)

File: lib/services/p2p/index.js [hunk 4]
```diff
@@ -332,8 +332,9 @@ P2P.prototype._setResourceFilter = function(filter, resource) {
 };
 
 P2P.prototype._startBcoin = function() {
+  const network = ['livenet', 'live', 'main', 'mainnet'].indexOf(this.node.network) !== -1? 'main' : 'testnet'
```
Comment (gabegattis) [inline lib/services/p2p/index.js]: please use jshint

File: lib/scaffold/default-config.js (hunk 1 above)
Comment (bitjson) [inline lib/scaffold/default-config.js]: You're right, that was a bad idea. We also need to figure out how to help people upgrade if they already have a config file. (Or at least show a warning if a config file from an older version exists.) I'll look at that now.

**Pull request merged**